crypten/mpc/mpc.py

```diff
@@ -218,6 +233,117 @@ def to(self, *args, **kwargs):
             self.share = share
             return self
 
+    def add(self, y):
+        return self.clone().add_(y)
+
+    def add_(self, y):
+        """added addition according to MAC"""
+
+        private = isinstance(y, MPCTensor)
+
+        if not private:
+            self._tensor += y
+            scaled_y = self._tensor.encoder.encode(y)
+            if self._mac is not None:
+                self._mac += self.alpha * scaled_y
+
+        else:
+            self._tensor += y._tensor
+            if self._mac is not None:
+                self._mac += y._mac
+
+        return self
+
+    def mul(self, y):
+        if self._mac is None:
+            if isinstance(y, MPCTensor):
+                y = y._tensor
+            self._tensor = self._tensor.mul_(y)
+        return self.clone().mul_(y)
+
+    def mul_(self, y):
+        """added multiplication according to MAC"""
+
+        private = isinstance(y, MPCTensor)
+
+        if not private:
+            self._tensor *= y
+            scaled_y = self._tensor.encoder.encode(y)
+            if self._mac is not None:
+                self._mac *= scaled_y
+        else:
+            raise NotImplementedError
+
+        return self
+
+    def __xor__(self, y):
+        """added xor according to MAC"""
+        result = self.clone()
+
+        private = isinstance(y, MPCTensor)
+
+        if isinstance(y, BinarySharedTensor):
+            broadcast_tensors = torch.broadcast_tensors(result.share, y.share)
+            result.share = broadcast_tensors[0].clone()
+        elif is_tensor(y):
+            broadcast_tensors = torch.broadcast_tensors(result.share, y)
+            result.share = broadcast_tensors[0].clone()
+
+        if not private:
+            result._tensor ^= y
+            if self._mac is not None:
+                result._mac ^= y._mac ^ (result.alpha & y)
+        else:
+            result._tensor ^= y._tensor
+            if self._mac is not None:
+                result._mac ^= y._mac
+
+        return result
+
+    def __and__(self, y):
+        """added and according to MAC"""
+        result = self.clone()
+
+        private = isinstance(y, MPCTensor)
+
```

Comment:
`__and__` works just like `mul`.

So, I think we can do the following:
```
if self._mac is None:
    if private:
        y = y._tensor
    self._tensor = self._tensor.mul_(y)
```
This will take care of the private `__and__` case for passive security.

(Note: this code is taken from _add_inplace_binary_passthrough_function on line 1394)